src/main/java/com/flansmod/common/guns/EntityBullet.java

```diff
@@ -632,11 +636,20 @@ else if (motionZ != 0)
                             showCrosshair = true;
                         }
                     }
-
+                    //double hit fix
                     PlayerBulletHit playerHit = (PlayerBulletHit) bulletHit;
-                    penetratingPower = playerHit.hitbox.hitByBullet(this, penetratingPower);
-                    if (FlansMod.DEBUG)
-                        worldObj.spawnEntityInWorld(new EntityDebugDot(worldObj, new Vector3f(posX + motionX * playerHit.intersectTime, posY + motionY * playerHit.intersectTime, posZ + motionZ * playerHit.intersectTime), 1000, 1F, 0F, 0F));
+                    boolean hasHit = false;
```

Comment:
Overall this looks good: But could you please remove this commit (for double hit fix) and add it as a separate pull request, if that's what you'd like to do?
Otherwise, it's ready to merge.